spec.html

```diff
@@ -49463,54 +49463,39 @@ <h1>
           </dl>
           <emu-alg>
             1. Let _errors_ be a new empty List.
+            1. [declared="remainingElementsCount"] NOTE: _remainingElementsCount_ starts at 1 instead of 0 to ensure _resultCapability_.[[Reject]] is only called once, even in the presence of a misbehaving *"then"* which calls the passed callback before the input iterator is exhausted.
             1. Let _remainingElementsCount_ be the Record { [[Value]]: 1 }.
             1. Let _index_ be 0.
             1. Repeat,
               1. Let _next_ be ? IteratorStepValue(_iteratorRecord_).
               1. If _next_ is ~done~, then
                 1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
                 1. If _remainingElementsCount_.[[Value]] = 0, then
-                  1. Let _error_ be a newly created *AggregateError* object.
-                  1. Perform ! DefinePropertyOrThrow(_error_, *"errors"*, PropertyDescriptor { [[Configurable]]: *true*, [[Enumerable]]: *false*, [[Writable]]: *true*, [[Value]]: CreateArrayFromList(_errors_) }).
-                  1. Return ThrowCompletion(_error_).
+                  1. Let _aggregateError_ be a newly created *AggregateError* object.
+                  1. Perform ! DefinePropertyOrThrow(_aggregateError_, *"errors"*, PropertyDescriptor { [[Configurable]]: *true*, [[Enumerable]]: *false*, [[Writable]]: *true*, [[Value]]: CreateArrayFromList(_errors_) }).
+                  1. Perform ? Call(_resultCapability_.[[Reject]], *undefined*, « _aggregateError_ »).
```

Comment:
Correctness of this change is a little subtle: it relies on the fact that we are in the `_next_ is ~done~` branch, which implies that `iteratorRecord.[[Done]] is *true*` in `Promise.any` (because of step 5.a in [IteratorStep](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-iteratorstep)), and hence taking step 8 in `Promise.any` is equivalent to calling `_resultCapability_.[[Reject]]` here and then taking step 9. However, I think it is indeed correct; just noting that for other readers.